content/developer/reference/guidelines.rst

```diff
@@ -999,17 +999,88 @@ More precise JS guidelines are detailed in the `github wiki  <https://github.com
+
+The name on the root element should follow the format `o-ComponentName`. And for children elements `o-ComponentName-partName` where:
+
+- `o-` is a must have Odoo prefix.
+- `ComponentName` is the PascalCase name of the current component/template.
+- `-` is a mandatory hyphen to separate the component name and the part name.
+- `partName` is the camelCase name of the child element referred by this `name` attribute.
+
+Attributes
+~~~~~~~~~~
+
+All the attributes of an element should be on the same line as the element itself. No character limit per line.
+
+.. note::
+    For extremely long lines, it is left to the appreciation of the reviewer to accept multi-line attributes within an element (but it is not mandatory).
+
+    In that case each attribute should be on its own line, with an indentation of 4 spaces relative to the element itself. The closing of the element (either `>` or `/>`) should be on its own line as well, with the same indentation as the corresponding opening character.
+
+The following order is applied for attributes:
+
+ #. `name`
+ #. `t-if`, `t-elif`, `t-else`
+ #. `t-foreach`, `t-as`
+ #. attributes specific to this type of element (eg. `type` for `<input>`, `src` for `<img>`)
+ #. `t-on-`
+ #. `class`
+ #. other generic attributes (eg. `title`, `aria-`)
+ #. `id`
+ #. OWL technical attributes (eg. `t-key`, `t-ref`)
+
+`t-att-` and `t-attf-` variations are ordered just after their static counter-part, in this order respectively.
+
+.. note:: `id` should not be used unless absolutely necessary (such as for making an anchor, or as the `for` target of a `label`), and then it should be uniquely generated.
+
+Style
+~~~~~
+
+Prefer using Bootstrap components and utilities over anything else, by applying their classes on the template. Custom style should only be used if absolutely necessary.
+
+Inside the `class` attribute itself, the following order should be respected:
+
+ #. Custom template class (only if necessary for adding custom style).
+ #. Bootstrap component classes (eg. `btn`, `card`).
+ #. Bootstrap layout classes (anything that changes the size or position of the element or its children, eg. `d-flex`, `container`, `mt-2`).
+ #. Bootstrap style classes (eg. `text-muted`, `bg-primary`).
+
+.. warning:: Before writing custom style you are expected to know all existing Bootstrap components, utilities, mixin and variables, as well as all of Odoo's extra components, utilities, mixin and variables.
+
+If necessary a `css` or `scss` file should be used, one per component/template. Inside that file, the style rules should be as flat as possible and target specific parts of the component/template.
+
+
+The naming guideline for targeting parts of the template with CSS classes is: `o-ComponentName-styleName` where
```

Comment:
as said above, this is horrible. The overhwelming convention in css world is to only use hypens and no capital letter.

Reply:
What is your suggestion for name then, all lower case and only hyphens?
With double hyphens to separate semantically different parts of the name or not?

Reply:
just lowercase and only hyphens. no need for double hyphens. Just something like `o-webclient` or `o-form-view` looks nice. If you really want to have a sub part, I guess it could be `o-form-view-header`. On that topic, I think we made a mistake of using the `o_` prefix in the past (it was a renaming of the `oe_` prefix, so a logical evolution). I am in favor of using `o-` from now on instead of `o_`